Removing AI models except cohere

rk-ansearch opened the pull request.

rk-ansearch committed:
Removing AI models except cohere

rk-ansearch committed:
removing option for model

Mael-Abgrall reviewed (changes requested):
Good

You can do further simplifications, add important features, and remove unused packages:

1. extract this into an ai package
2. Move the package @aws-sdk/client-bedrock-runtime to the newly created ai package
3. Remove @huggingface/transformers
4. (optional) Add analytics and usage reporting with Posthog <https://posthog.com/docs/ai-engineering/observability>

rk-ansearch committed:
moving into packages

rk-ansearch committed:
moving into packages

Mael-Abgrall committed:
Add Drizzle ORM and PostgreSQL database support (#163)

* Add Drizzle ORM and PostgreSQL database support

- Integrated Drizzle ORM for database management
- Added PostgreSQL configuration and schemas
- Updated database vector search to use PostgreSQL
- Added environment type definitions for database configuration
- Updated package dependencies to support database operations
- Configured Drizzle for database migrations and studio

* Fts document migration (#164)

* Adding documents Table

* saving into the table

* retrieve document

* Saving content for FTS

* adding content for FTS

* modifying delete function

* removing contents table

* FTS Method

* removing documentID in FTS

* Full Text search

* renaming function

* Refactor database structure and search functionality

- Updated drizzle configuration to point to the new database schema file.
- Added new database configuration files for PostgreSQL.
- Implemented document management functions including adding, deleting, and updating documents in the database.
- Introduced chunk management for search functionality, allowing for efficient storage and retrieval of document chunks.
- Removed deprecated FTS methods and integrated new search chunk methods.
- Added integration and unit tests for the new document and chunk management functionalities.
- Set up test environment with database initialization and cleanup procedures.

---------

Co-authored-by: Mael Abgrall <172921326+Mael-Abgrall@users.noreply.github.com>

---------

Co-authored-by: Rohith <173175567+rk-ansearch@users.noreply.github.com>

Mael-Abgrall committed:
Add Drizzle ORM and PostgreSQL database support (#163)

* Add Drizzle ORM and PostgreSQL database support

- Integrated Drizzle ORM for database management
- Added PostgreSQL configuration and schemas
- Updated database vector search to use PostgreSQL
- Added environment type definitions for database configuration
- Updated package dependencies to support database operations
- Configured Drizzle for database migrations and studio

* Fts document migration (#164)

* Adding documents Table

* saving into the table

* retrieve document

* Saving content for FTS

* adding content for FTS

* modifying delete function

* removing contents table

* FTS Method

* removing documentID in FTS

* Full Text search

* renaming function

* Refactor database structure and search functionality

- Updated drizzle configuration to point to the new database schema file.
- Added new database configuration files for PostgreSQL.
- Implemented document management functions including adding, deleting, and updating documents in the database.
- Introduced chunk management for search functionality, allowing for efficient storage and retrieval of document chunks.
- Removed deprecated FTS methods and integrated new search chunk methods.
- Added integration and unit tests for the new document and chunk management functionalities.
- Set up test environment with database initialization and cleanup procedures.

---------

Co-authored-by: Mael Abgrall <172921326+Mael-Abgrall@users.noreply.github.com>

---------

Co-authored-by: Rohith <173175567+rk-ansearch@users.noreply.github.com>

Mael-Abgrall committed:
Add Drizzle ORM and PostgreSQL database support (#163)

* Add Drizzle ORM and PostgreSQL database support

- Integrated Drizzle ORM for database management
- Added PostgreSQL configuration and schemas
- Updated database vector search to use PostgreSQL
- Added environment type definitions for database configuration
- Updated package dependencies to support database operations
- Configured Drizzle for database migrations and studio

* Fts document migration (#164)

* Adding documents Table

* saving into the table

* retrieve document

* Saving content for FTS

* adding content for FTS

* modifying delete function

* removing contents table

* FTS Method

* removing documentID in FTS

* Full Text search

* renaming function

* Refactor database structure and search functionality

- Updated drizzle configuration to point to the new database schema file.
- Added new database configuration files for PostgreSQL.
- Implemented document management functions including adding, deleting, and updating documents in the database.
- Introduced chunk management for search functionality, allowing for efficient storage and retrieval of document chunks.
- Removed deprecated FTS methods and integrated new search chunk methods.
- Added integration and unit tests for the new document and chunk management functionalities.
- Set up test environment with database initialization and cleanup procedures.

---------

Co-authored-by: Mael Abgrall <172921326+Mael-Abgrall@users.noreply.github.com>

---------

Co-authored-by: Rohith <173175567+rk-ansearch@users.noreply.github.com>

rk-ansearch committed:
rebasing
